src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DocumentHighlightHandler.cs

```diff
@@ -66,14 +77,21 @@ public async Task<DocumentHighlight[]> HandleRequestAsync(DocumentHighlightParam
                 throw new ArgumentNullException(nameof(clientCapabilities));
             }
 
+            _logger.LogInformation($"Starting request for {request.TextDocument.Uri}.");
+
             if (!_documentManager.TryGetDocument(request.TextDocument.Uri, out var documentSnapshot))
             {
+                _logger.LogWarning($"Failed to find document {request.TextDocument.Uri}.");
                 return null;
             }
 
-            var projectionResult = await _projectionProvider.GetProjectionAsync(documentSnapshot, request.Position, cancellationToken).ConfigureAwait(false);
+            var projectionResult = await _projectionProvider.GetProjectionAsync(
+                documentSnapshot,
+                request.Position,
+                cancellationToken).ConfigureAwait(false);
             if (projectionResult == null)
             {
+                _logger.LogInformation($"Failed to find projection in {request.TextDocument.Uri}.");
```

Comment:
I wonder if instead of adding this log to all of our handlers if we should expand our projection provider to do this

Reply:
I was debating that as well. I went this route to account for the other (expected?) cases when we can't find projections and don't want to bother with logging that. If that's not a concern, I can move this over to the projection provider.

Reply:
I don't think it's a concern

Reply:
https://github.com/dotnet/aspnetcore-tooling/pull/3129/commits/fcd7b47ab787a92894004a247e9d76fa6656b1f0